ui/src/components/VideoOverlay.tsx

```diff
@@ -398,124 +399,126 @@ export function PointerLockBar({ show }: PointerLockBarProps) {
-          await sleep(1000);
-          window.location.reload();
+        // URL constructor handles relative paths, protocol-relative URLs, and absolute URLs
+        // Relative path → resolves against origin
+        // new URL("/device/status", "http://192.168.1.77").href
+        // // → "http://192.168.1.77/device/status"
+
+        // // Protocol-relative URL → uses protocol from base, host from URL
+        // new URL("//192.168.1.100/device/status", "http://192.168.1.77").href
+        // // → "http://192.168.1.100/device/status"
+
+        // // Fully qualified URL → base is ignored entirely
+        // new URL("http://192.168.1.100/device/status", "http://192.168.1.77").href
+        // // → "http://192.168.1.100/device/status"
+        const healthUrl = new URL(postRebootAction.healthCheck, window.location.origin).href;
+        const res = await fetch(healthUrl, { signal: controller.signal });
+        if (res.ok) {
+          clearInterval(intervalId);
+          const targetUrl = new URL(postRebootAction.redirectTo, window.location.origin).href;
+          await redirectTo(targetUrl);
         }
       } catch (err) {
-        // Ignore errors - they're expected while device is rebooting
-        // Only log if it's not an abort error
-        if (err instanceof Error && err.name !== 'AbortError') {
-          console.debug('Error checking post-reboot health:', err);
+        if (err instanceof Error && err.name !== "AbortError") {
+          console.debug("Health check failed:", err.message);
         }
       } finally {
-        clearTimeout(timeoutId);
-        isFetchingRef.current = false;
+        clearTimeout(timeout);
+        isCheckingRef.current = false;
       }
     };
 
-    // Start interval (check every 2 seconds)
-    const intervalId = setInterval(checkPostRebootHealth, 2000);
+    const intervalId = setInterval(checkHealth, 2000);
+    checkHealth();
 
-    // Also check immediately
-    checkPostRebootHealth();
-
-    // Cleanup on unmount or when dependencies change
     return () => {
       clearInterval(intervalId);
-      if (abortControllerRef.current) {
-        abortControllerRef.current.abort();
+      abortControllerRef.current?.abort();
+      isCheckingRef.current = false;
+    };
+  }, [show, postRebootAction, hasSeenDisconnect, redirectTo]);
+
+  // Cloud mode: wait for WebRTC reconnection via RPC, then redirect with versioned URL
+  useEffect(() => {
+    if (isOnDevice || !postRebootAction || !deviceId || !show || !hasSeenDisconnect) return;
+
+    let cancelled = false;
+
+    const waitForReconnectAndRedirect = async () => {
+      if (isCheckingRef.current) return;
+      isCheckingRef.current = true;
+
+      try {
+        // getLocalVersion internally waits for RTC channel to be ready
+        const { appVersion } = await getLocalVersion();
+        if (cancelled) return;
```

Comment:
_note to self_: do this second so that during the `await` if it gets cancelled...